packages/api-bindings/src/graphql/utils/publication.ts

```diff
@@ -93,8 +98,156 @@ export type PublicationOwnedByMeFragment = Overwrite<
   { profile: Overwrite<ProfileFragment, { ownedByMe: true }> }
 >;
 
-export const isPublicationOwnedByMe = (
+export function isPublicationOwnedByMe(
   publication: PublicationFragment,
-): publication is PublicationOwnedByMeFragment => {
+): publication is PublicationOwnedByMeFragment {
   return publication.profile.ownedByMe;
+}
+
+export function createCollectRequest(
+  publication: PublicationFragment,
+  activeProfile: ProfileFragment,
+): CollectRequest {
+  switch (publication.collectModule.__typename) {
+    case 'FreeCollectModuleSettings':
+      return {
+        profileId: activeProfile.id,
+        kind: TransactionKind.COLLECT_PUBLICATION,
+        publicationId: publication.id,
+        publicationType: getPublicationType(publication),
+        type: CollectType.FREE,
+      };
+    case 'FeeCollectModuleSettings':
+    case 'LimitedFeeCollectModuleSettings':
+    case 'TimedFeeCollectModuleSettings':
+    case 'LimitedTimedFeeCollectModuleSettings':
+      return {
+        profileId: activeProfile.id,
+        kind: TransactionKind.COLLECT_PUBLICATION,
+        publicationId: publication.id,
+        publicationType: getPublicationType(publication),
+        type: CollectType.PAID,
+        fee: {
+          amount: erc20Amount({ from: publication.collectModule.amount }),
+          contractAddress: publication.collectModule.contractAddress,
+        },
+      };
+    case 'RevertCollectModuleSettings':
+    default:
+      never(
+        `Cannot collect publication with "${
+          publication.collectModule.__typename as string
```

Comment:
is this because some times it's nullable? I think we can include it in the corresponding GQL fragment definition and it will not longer be `Maybe<string>`?

Reply:
We always are returned the `collectModule`, this default case is saying if it isn't any of the already listed cases, throw an error when trying to create a collection request because we don't support the collect module.

I have removed the default case in favour of an exhaustive switch check here.